### ISSUE TITLE
NullPointerException on matcher expression

### ISSUE THREAD

**Issue by tonysparks** [schibsted/jslt#41]
This expression throws a NullPointerException:

```json 
{ "a": "b" }
+
{
  "type" : "Anonymized-View",
  * : .
} 
```

Input:
```json
{ "test" : "t" }
```

I would expect the following result:
```json
{"type":"Anonymized-View","test":"t","a":"b"}
```

**Comment by larsga** [comment]
I've been a little lax with the formal definition of what `* : .` means. I'll do the job on this.

**Comment by larsga** [comment]
When considering the object matcher more deeply, also consider the pipe operator (issue #40).

**Comment by larsga** [comment]
This bug was fixed in commit 9da20ab
